lldb/source/Target/StackID.cpp

```diff
@@ -73,16 +73,41 @@ bool lldb_private::operator!=(const StackID &lhs, const StackID &rhs) {
   return lhs_scope != rhs_scope;
 }
 
-bool StackID::IsYounger(const StackID &lhs, const StackID &rhs,
-                        Process &process) {
+// BEGIN SWIFT
+enum class HeapCFAComparisonResult { Younger, Older, NoOpinion };
+/// If at least one of the stack IDs (lhs, rhs) is a heap CFA, perform the
+/// swift-specific async frame comparison. Otherwise, returns NoOpinion.
+static HeapCFAComparisonResult
+IsYoungerHeapCFAs(const StackID &lhs, const StackID &rhs, Process &process) {
+  const bool lhs_cfa_on_stack = lhs.IsCFAOnStack(process);
+  const bool rhs_cfa_on_stack = rhs.IsCFAOnStack(process);
+  if (lhs_cfa_on_stack && rhs_cfa_on_stack)
+    return HeapCFAComparisonResult::NoOpinion;
+
   // FIXME: rdar://76119439
   // At the boundary between an async parent frame calling a regular child
   // frame, the CFA of the parent async function is a heap addresses, and the
   // CFA of concrete child function is a stack address. Therefore, if lhs is
   // on stack, and rhs is not, lhs is considered less than rhs, independent of
   // address values.
-  if (lhs.IsCFAOnStack(process) && !rhs.IsCFAOnStack(process))
+  if (lhs_cfa_on_stack && !rhs_cfa_on_stack)
+    return HeapCFAComparisonResult::Younger;
+  return {};
```

Comment:
should this be `NoOpinion`?

(I'm not sure what `{}` evaluates to for an enum, but I presume the first case)

Reply:
Oooops.